lms/djangoapps/course_goals/tests/test_api.py

```diff
@@ -0,0 +1,66 @@
+
+"""
+Unit tests for course_goals.api methods.
+"""
+from django.contrib.auth.models import User
+from django.core.urlresolvers import reverse
+from lms.djangoapps.course_goals.models import CourseGoal
+from rest_framework.test import APIClient
+from student.models import CourseEnrollment
+from track.tests import EventTrackingTestCase
+from xmodule.modulestore.tests.django_utils import SharedModuleStoreTestCase
+from xmodule.modulestore.tests.factories import CourseFactory
+
+TEST_PASSWORD = 'test'
+
+
+class TestCourseGoalsAPI(EventTrackingTestCase, SharedModuleStoreTestCase):
+    """
+    Testing the Course Goals API.
+    """
+
+    def setUp(self):
+        # Create a course with a verified track
+        super(TestCourseGoalsAPI, self).setUp()
+        self.course = CourseFactory.create()
+
+        self.user = User.objects.create_user('john', 'lennon@thebeatles.com', 'password')
+        CourseEnrollment.enroll(self.user, self.course.id)
+
+        self.client = APIClient(enforce_csrf_checks=True)
+        self.client.login(username=self.user.username, password=self.user.password)
+        self.client.force_authenticate(user=self.user)
+
+        self.apiUrl = reverse('course_goals_api:v0:course_goal-list')
+
+    def test_add_valid_goal(self):
```

Comment:
I could imagine a single ddt with lots of combinations for various combos of goal, user, and course_key.  Unit tests are where you should test happy and sad paths.  We also want to ensure that you can't set a goal for a different user which may or may not be a separate test.

Reply:
FYI: @HarryRein: You never added the test for permissions (i.e. I can't set someone else's goal).  It would be great if this were done as part of the update tests.